src/Clapi/Valuespace.hs

```diff
@@ -414,6 +413,8 @@ data ValidationErr
   | ProgrammingErr String
   | MissingChild Seg
   | ExtraChild Seg
+  | RefTargetNotFound Path
+  | RefTargetTypeErr Path TypeName TypeName
```

Comment:
Might be nice to use record syntax for some of these errors, even if just so that naming the fields is a hint as to which args are which!